src/app/components/dashboard/dashboard.controller.js

```diff
@@ -24,5 +27,47 @@
       });
     });
 
+    $scope.removeFavorite = function (list) {
+      $scope.currentUser.favoriteLists.splice($scope.currentUser.favoriteLists.indexOf(list), 1);
+
+      User.update($scope.currentUser, function () {
+        alertService.add('success', gettextCatalog.getString('This list was removed from your favorites.'));
+        $scope.setCurrentUser($scope.currentUser);
+      });
+    };
+
+    $scope.leaveList = function (list) {
```

Comment:
It's correct. In some refactoring, it would be good to put these functions in a list service so that we can reuse them, but for now it's fine.